### ISSUE TITLE
Cannot properly use groupify when using prefixed table names

### ISSUE THREAD

**Issue by Tuxii** [dwbutler/groupify#57]
Hi,

Suppose I have this Groupify configuration 

```ruby
Groupify.configure do |config|
  # Configure the default group class name.
  # Defaults to 'Group'
  config.group_class_name = 'Company::Group'

  # Configure the default group membership class name.
  # Defaults to 'GroupMembership'
  config.group_membership_class_name = 'Company::GroupMembership'
end
```

And I have my groupify tables prefixed accordingly : 

```ruby
class GroupifyMigration < ActiveRecord::Migration
  def change
    create_table :company_groups do |t|
      t.string     :type
    end

    create_table :company_group_memberships do |t|
      t.references :member, polymorphic: true, index: true, null: false
      t.references :group, polymorphic: true, index: true

      # The named group to which a member belongs (if using)
      t.string     :group_name, index: true

      # The membership type the member belongs with
      t.string     :membership_type

      t.timestamps
    end
  end
end
```

If I try to do the following command : 

```ruby
Company::User.shares_any_group Company::User.last
```

It result in the following query : 

```sql
Company::User Load (0.2ms)  SELECT  "company_users".* FROM "company_users" ORDER BY "company_users"."id" DESC LIMIT ?  [["LIMIT", 1]]
DEPRECATION WARNING: uniq is deprecated and will be removed from Rails 5.1 (use distinct instead) (called from irb_binding at (irb):20)
  Company::Group Load (0.2ms)  SELECT DISTINCT "company_groups".* FROM "company_groups" INNER JOIN "company_group_memberships" ON "company_groups"."id" = "company_group_memberships"."group_id" WHERE "company_group_memberships"."member_id" = ? AND "company_group_memberships"."member_type" = ? AND "company_group_memberships"."group_type" = ?  [["member_id", 5], ["member_type", "Company::User"], ["group_type", "Company::Group"]]
DEPRECATION WARNING: uniq is deprecated and will be removed from Rails 5.1 (use distinct instead) (called from irb_binding at (irb):20)
  Company::User Load (0.5ms)  SELECT DISTINCT "company_users".* FROM "company_users" INNER JOIN "company_group_memberships" ON "company_group_memberships"."member_id" = "company_users"."id" AND "company_group_memberships"."member_type" = ? WHERE "group_memberships"."group_id" = 2  [["member_type", "Company::User"]]
```

Which raises in a `ActiveRecord::StatementInvalid: SQLite3::SQLException: no such column: group_memberships.group_id:` error because in the request, it references the `group_memberships` table and not the `company_group_memberships` table. 

Also if I try this :

```ruby
Company::User.last.named_groups << :admin
```
I get an `ActiveRecord::RecordInvalid: Validation failed: Group must exist` error. There again, it would seem Groupify does not use the Company::Group model

Here is a test repository to reproduce this behavior : https://github.com/Tuxii/namespace_groupify

So in other words, Groupify allows to rename its models via the config initializer, but that does not mean you can rename the table as well right ?

**Comment by dwbutler** [comment]
Thanks, I'm working on a fix for this now. This affects any custom Group Membership model, not just namespaced models.

Named Groups don't use a Group model at all. It only uses the Group Membership model. So that probably isn't working for the same reason.
